credentials/apps/badges/signals/handlers.py

```diff
@@ -33,4 +39,18 @@ def event_handler(sender, signal, **kwargs):
     logger.debug(f"Received signal {signal}")
 
     # NOTE (performance): all consumed messages from event bus trigger this.
-    process(signal, sender=sender, **kwargs)
\ No newline at end of file
+    process(signal, sender=sender, **kwargs)
+
+
+@receiver(BADGE_PROGRESS_COMPLETE)
+def listen_for_completed_badge(sender, username, badge_template_id, **kwargs):  # pylint: disable=unused-argument
+    badge_template = get_badge_template_by_id(badge_template_id)
+
+    if badge_template is None:
+        return
+
+    if badge_template.origin == 'openedx':
```

Comment:
Note for future: here we'd want to make badge templates to be responsible for their type of user credentials:

`badge_template.award(username)` 
`badge_template.revoke(username)`

- each badge template type (with a specific origin) manages its type of user credential;